File: BackEnd/budget-service/src/main/java/com/teja/budget_service/service/BudgetService.java
package com.teja.budget_service.service;

import com.teja.budget_service.model.BudgetModel;
import com.teja.budget_service.model.PageableBudgets;

public interface BudgetService {

    BudgetModel createBudget(BudgetModel budgetModel);

    PageableBudgets getBudgets(int page, int size);

    BudgetModel updateBudget(String budgetId, BudgetModel budgetModel);

    void deleteBudget(String budgetId);

}

File: FrontEnd/personal-expense-tracker-ui/src/app/features/budget/services/budget.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment';
import { budget } from '../model/budget.model';

@Injectable({
  providedIn: 'root'
})
export class BudgetService {

  constructor(private http: HttpClient) { }

  getBudgets(page:number,size:number) {
    return this.http.get<any>(environment.budgetServiceURL);
  }

  createBudget(budget: budget) {
    budget.userId = "testUser";
    return this.http.post<any>(environment.budgetServiceURL, budget);
  }

  updateBudget(budget: any) {
    return this.http.put<any>(environment.budgetServiceURL, budget);
  }

  deleteBudget(id: any) {
    return this.http.delete<any>(`${environment.budgetServiceURL}/${id}`);
  }
}
